### PR TITLE
[Feature:CourseMaterials] Button to clear "NEW" badge 

### PR DESCRIPTION
**Pull request by hayeonohh**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
New materials posted in the course materials page show up with a "NEW" badge next to them. However, there is no way to mark all the course materials as seen at once. The only way to mark them as viewed is to download them which can be an inconvenience if there are a lot of small files that don't need to be viewed.

### What is the new behavior?
There is now a button that marks all of the course materials as seen, which gets rid of the "NEW" badge next to them. 
image

image

### Other information?
Closes issue #9941

### CODE REVIEW

File: site/app/controllers/course/CourseMaterialsController.php [hunk 1]
```diff
@@ -104,6 +104,20 @@ public function markViewed(): JsonResponse {
...
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/course_materials/viewAll", methods={"POST"})
+     */
+    public function setAllViewed(): JsonResponse {
+        $cms = $this->core->getCourseEntityManager()->getRepository(CourseMaterial::class)
+            ->findAll();
+        foreach ($cms as $cm) {
+            $cm_access = new CourseMaterialAccess($cm, $this->core->getUser()->getId(), $this->core->getDateTimeNow());
+            $cm->addAccess($cm_access);
```
Comment (williamjallen) [inline site/app/controllers/course/CourseMaterialsController.php:115]: Is this also going to add an access for hidden course materials?

File: site/public/js/server.js [hunk 2]
```diff
@@ -870,6 +870,35 @@ function markViewed(ids, redirect) {
...
+    let data = new FormData();
+    data.append("csrf_token", csrfToken);
+    $.ajax({
+        url: buildCourseUrl(['course_materials', 'viewAll']), 
+        type: "POST",
+        data: data,
+        contentType: false,
+        processData: false,
+        success: function(response) {
+            console.log("Success:", response);
```
Comment (williamjallen) [inline site/public/js/server.js:888]: Should this console.log be removed?

File: site/public/js/server.js [hunk 3]
```diff
@@ -870,6 +870,35 @@ function markViewed(ids, redirect) {
...
+            // Optional: Redirect or perform some action on success
+            if (redirect) {
+                window.location.href = redirect;
```
Comment (williamjallen) [inline site/public/js/server.js:891]: Why are you executing an AJAX request and then promptly reloading the page?  You could handle the redirect on the server side.

**Review by VVVITH — APPROVED** [review]
The changes William required haven't responds more than 2 weeks. So I work on it.
I make some changes:
1. Make a block for hidden or nonreleased homework, the viewed button should not affect them.
2. Fix the code error about var-> const.
3. Rename the button and the functions to be more clear.

Need more reviews. Since I made the main changes, my approve should not be count.
You can test this by uploading more files and set them to unrelease or hidden or both.

**Review by cjreed121 — APPROVED** [review]

**Pull request merged**